manager-parent/manager-core/src/main/java/com/redhat/service/smartevents/manager/core/services/ShardServiceImpl.java

```diff
@@ -30,7 +35,25 @@ public void init() {
          * Fetch the list of authorized shards at startup.
          * This needs to be changed when admin api will include the CRUD of a shard at runtime.
          */
-        shards = shardDAO.listAll().stream().map(Shard::getId).collect(Collectors.toSet());
+        shardIds = shardDAO.listAll().stream().map(Shard::getId).collect(Collectors.toSet());
+
+        // Align Shard router canonical hostname database entry with value provided by properties - https://issues.redhat.com/browse/MGDOBR-1241
+        List<Shard> shards = shardDAO.listAll();
+        if (shards.size() == 1) {
+            Shard shard = shards.get(0);
+            String routerCanonicalHostname = shard.getRouterCanonicalHostname();
+            if (!shardRouterCanonicalHostnameFromProperties.equals(routerCanonicalHostname)) {
+                LOGGER.warn("Shard router canonical hostname '{}' doesn't match expected value '{}' from properties file, replacing hostname to match properties.", routerCanonicalHostname,
+                        shardRouterCanonicalHostnameFromProperties);
+                shard.setRouterCanonicalHostname(shardRouterCanonicalHostnameFromProperties);
+                persist(shard);
+            }
+        }
+    }
+
+    @Transactional
```

Comment:
@sutaakar If you move the `@Transactional` annotation onto the `init` method then you can delete this method.

The `ShardDAO.listAll()` method will load the `Shard` entity into the JPA `PersistenceContext` (move the entity into the "managed" lifecycle state). Your call to `Shard.setRouterCanonicalHostname` will mark the entity as "dirty" and then Hibernate will automatically flush the changes to the DB when the transaction commits. 

There's a high-level overview here of entity lifecycles and the PersistenceContext: https://www.baeldung.com/hibernate-entity-lifecycle#:~:text=Every%20Hibernate%20entity%20naturally%20has,able%20to%20use%20Hibernate%20properly.

cheers,

Rob

Reply:
@robpblake Unfortunately `@Transactional` is not compatible with `@PostConstruct`. As a result when both annotations are used on same method then transaction doesn't seem to be opened - changes are not persisted.

Reply:
@sutaakar hmm...shame. Thanks for trying it though.

Reply:
we could do a JPA update then `where name != shard.getRouterCanonicalHostname()`

There's also `update` on Panache but I've never used it